### PR TITLE
feat: mavenPropertyVersions regexManager

### PR DESCRIPTION
**Pull request by rombert**
## Changes

Adds a new regexManager that can be used for properties defined in pom.xml Maven files.

## Context

https://github.com/renovatebot/renovate/discussions/21495

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/presets/internal/regex-managers.ts [hunk 1]
```diff
@@ -44,6 +44,21 @@ export const presets: Record<string, Preset> = {
...
+  mavenPropertyVersions: {
+    description: 'Update `*.version` properties in `pom.xml` files.',
+    regexManagers: [
+      {
+        customType: 'regex',
+        datasourceTemplate:
+          '{{#if datasource}}{{{datasource}}}{{else}}maven{{/if}}',
+        fileMatch: ['^pom\\.xml$'],
+        matchStrings: [
+          '<!--\\s?renovate:( datasource=(?<datasource>.*?))? depName=(?<depName>.*?)\\s?(?: (?:lookupName|packageName)=(?<packageName>[^\\s]+?))?(?: versioning=(?<versioning>[a-z-0-9]+?))?\\s+-->\\s+<.*?\\.version>(?<currentValue>.*?)<\\/.*?\\.version>',
```
Comment (viceice) [inline lib/config/presets/internal/regex-managers.ts:56]:
```suggestion
          '<!--\\s?renovate:( datasource=(?<datasource>.*?))? depName=(?<depName>.*?)\\s?(?: packageName=(?<packageName>[^\\s]+?))?(?: versioning=(?<versioning>[a-z-0-9]+?))?\\s+-->\\s+<.*?\\.version>(?<currentValue>.*?)<\\/.*?\\.version>',
```
lookup name is long deprecated, so it shouldn't be used in new official presets

**Review by viceice — COMMENTED** [review]
isn't this something the maven manager should support out of the box? 🤔

**Comment by rombert** [comment]
> isn't this something the maven manager should support out of the box? 🤔

Good question. These properties are actually consumed by other plug-ins, which are not visible to the maven dependency manager. Not sure if they could be. Two examples I am aware of:

- pom property at https://github.com/apache/sling-org-apache-sling-starter/blob/ee7eb41337ec4510646d95f1fa951fa8e0b344d0/pom.xml#L51 ; consumed by the the OSGi feature model at https://github.com/apache/sling-org-apache-sling-starter/blob/ee7eb41337ec4510646d95f1fa951fa8e0b344d0/src/main/features/boot.json#L176 . I contributed the osgi manager to renovate, if you think we can make this work in a different way out-of-the-box I can try
- pom property at https://github.com/apache/sling-org-apache-sling-starter/blob/ee7eb41337ec4510646d95f1fa951fa8e0b344d0/pom.xml#L68 ; consumed by the docker-maven-plugin at https://github.com/apache/sling-org-apache-sling-starter/blob/ee7eb41337ec4510646d95f1fa951fa8e0b344d0/pom.xml#L288  ; this could also be extended in case the maven manager starts being aware of other configurations except the "dependency"/"dependencyManagement" ones.

File: lib/config/presets/internal/regex-managers.ts [hunk 2]
```diff
@@ -44,6 +44,21 @@ export const presets: Record<string, Preset> = {
...
+  mavenPropertyVersions: {
+    description: 'Update `*.version` properties in `pom.xml` files.',
+    regexManagers: [
+      {
+        customType: 'regex',
+        datasourceTemplate:
+          '{{#if datasource}}{{{datasource}}}{{else}}maven{{/if}}',
+        fileMatch: ['^pom\\.xml$'],
+        matchStrings: [
+          '<!--\\s?renovate:( datasource=(?<datasource>.*?))? depName=(?<depName>.*?)\\s?(?: packageName=(?<packageName>[^\\s]+?))?(?: versioning=(?<versioning>[a-z-0-9]+?))?\\s+-->\\s+<.*?\\.version>(?<currentValue>.*?)<\\/.*?\\.version>',
```
Comment (Churro) [inline lib/config/presets/internal/regex-managers.ts:56]:
```suggestion
          '<!--\\s?renovate:( datasource=(?<datasource>[a-z-.]+?))? depName=(?<depName>[^\\s]+?)(?: packageName=(?<packageName>[^\\s]+?))?(?: versioning=(?<versioning>[a-z-0-9]+?))?\\s+-->\\s+<.+\\.version>(?<currentValue>.+)<\\/.+\\.version>',
```

* `datasource`: other regex manager presets also limit it to `[a-z-.]+?`
  * https://github.com/renovatebot/renovate/pull/24001
* `depName`: same format as `packageName`, so should be matched analogously
* `<.version></.version>` would also be matched but this is not legit 
  * refine to `<.+\.version>(?<currentValue>.+)<\/.+\.version>`

**Review by Churro — CHANGES_REQUESTED** [review]
> isn't this something the maven manager should support out of the box? 🤔

It does: https://github.com/renovatebot/renovate/blob/7779ebdb8877cff6b1abb2ce8f3c02450ce05cd7/lib/modules/manager/maven/extract.ts#L275-L287

I guess it's really about those special cases @rombert mentioned, where properties are defined but not consumed by dependencies.

**Review by Churro — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**